### PR TITLE
fix(worker): Use Bridge data for Delay step during stateful execution

### PR DESCRIPTION
**Pull request by rifont**
### What changed? Why was the change needed?

* Bridge response data was not being used when executing a V2 Workflow with a delay step, resulting in all delay steps in V2 having a delay of 0 during execution. This change simplifies handling of the job metadata and ensures that bridge response data is used for the job

### Screenshots

_V1 Workflow with delay_
image

_V2 Stateless (Local Studio) workflow with delay_
image

_V2 Stateful (Cloud Dashboard) workflow with delay_
image

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/worker/src/app/workflow/usecases/add-job/add-delay-job.usecase.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export class AddDelayJob {
 
     try {
       delay = this.computeJobWaitDurationService.calculateDelay({
-        stepMetadata: data.step.bridgeUrl ? data.digest : data.step.metadata,
+        stepMetadata: data.step.metadata,
```
Comment (rifont) [inline apps/worker/src/app/workflow/usecases/add-job/add-delay-job.usecase.ts:42]: `AddDelayJob` no longer cares about v1 vs v2 or stateful vs stateless, that responsibility is with `AddJob` now

File: apps/worker/src/app/workflow/usecases/add-job/add-job.usecase.ts [hunk 2]
```diff
@@ -143,7 +143,7 @@ export class AddJob {
     }
 
     if (job.type === StepTypeEnum.DELAY) {
-      delayAmount = await this.handleDelay(command, filterVariables, delayAmount);
+      delayAmount = await this.handleDelay(command, filterVariables);
```
Comment (rifont) [inline apps/worker/src/app/workflow/usecases/add-job/add-job.usecase.ts:146]: `delayAmount` passed in here was never used.

File: apps/worker/src/app/workflow/usecases/add-job/add-job.usecase.ts [hunk 3]
```diff
@@ -199,21 +216,16 @@ export class AddJob {
...
 
-    const job = await this.updateJob(response, command);
-
-    // Update the job digest directly to avoid an extra database call
-    command.job.digest = { ...command.job.digest, ...job } as IWorkflowStepMetadata;
-
     return response;
   }
 
-  private async updateJob(response: ExecuteOutput<DigestOutput>, command: AddJobCommand) {
```
Comment (rifont) [inline apps/worker/src/app/workflow/usecases/add-job/add-job.usecase.ts:210]: Changes below here are just renaming to make for a more accurate description of the method.

File: apps/worker/src/app/workflow/usecases/add-job/add-job.usecase.ts [hunk 4]
```diff
@@ -170,13 +170,30 @@ export class AddJob {
...
+        ...command,
+        job: {
+          ...command.job,
+          step: {
+            ...command.job.step,
+            metadata,
+          },
+        },
+      })
+    );
```
Comment (rifont) [inline apps/worker/src/app/workflow/usecases/add-job/add-job.usecase.ts:196]: No longer mutating the data, constructing a new command now.

**Review by scopsy — APPROVED** [review]
💪

**Pull request merged**